src/hamlib_keyer.c

```diff
@@ -24,11 +24,13 @@
 #include "hamlib_keyer.h"
 
 bool rig_has_send_morse() {
+   // SAFETY: should not be written after init
    return (my_rig->caps->send_morse != NULL);
 }
 
 bool rig_has_stop_morse() {
 #if HAMLIB_VERSION >= 400
+    // SAFETY: should not be written after init
```

Comment:
I'm not sure: what should not be written?

Reply:
I am referencing the my_rig->caps field.

Reply:
I see. Don't we only read it?

Reply:
Yes, but technically, some other thread could update it while it is being read resulting in undefined behavior.

Reply:
True. But we as a hamlib client shall not mess around rig caps in general. Whether hamlib updates them at runtime (highly unlikely) that is out of our control anyway.
Wouldn't a note reminding that these fields are evaluated on init be sufficient? Or should these be also read and cached in `init_tlf_rig()` a bit like `rigptt`?

Reply:
I think having global flags for has_send_morse and has_stop_morse out of hamlib's reach is the best option.

Reply:
I have pushed a patch with flags that are set on init.